File: chapter-3_numerical-algorithms/javascript/lcm/test-lcm.js
'use strict';

var lcm = require('./lcm');
var gcd = require('../gcd/euclidian-gcd');

console.log("gcd(75, 25): " + gcd(75, 25));
console.log("lcm(75, 25): " + lcm(75, 25));

console.log("\ngcd(78, 66): " + gcd(78, 66));
console.log("gcd(78, 66): " + lcm(78, 66));

console.log("\ngcd(1500, 32589): " + gcd(1500, 32589));
console.log("gcd(1500, 32589): " + lcm(1500, 32589));

console.log("\ngcd(56789, 12344567): " + gcd(56789, 12344567));
console.log("lcm(56789, 12344567): " + lcm(56789, 12344567));

console.log("\ngcd(10000, 20000): " + gcd(10000, 20000));
console.log("lcm(10000, 20000): " + lcm(10000, 20000));

console.log("\ngcd(12, 15): " + gcd(12, 15));
console.log("lcm(12, 15): " + lcm(12, 15));

File: chapter-3_numerical-algorithms/python/lcm/test_lcm.py
import sys
sys.path.insert(0, '../gcd/')

from lcm import lcm
from euclidian_gcd import gcd

print(gcd(25, 75))
print(lcm(25, 75))

print(gcd(78, 66))
print(lcm(78, 66))

print(gcd(1500, 32589))
print(lcm(1500, 32589))

print(gcd(56789, 12344567))
print(lcm(56789, 12344567))

print(gcd(12, 15))
print(lcm(12, 15))
